### PR TITLE
Implement grid search for all classifiers.

### PR DESCRIPTION
**Pull request by dtimm**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
 classifiers, \
     train_times, \
     pred_times, \
     f1_trains, \
-    f1_tests = runTests(X_test, X_train, y_test, y_train)
+    f1_tests, \
+    gs_times = runTests(X_test, X_train, y_test, y_train)
 
-writeToCsv(classifiers, train_times, pred_times, f1_trains, f1_tests)
+writeToCsv(classifiers, train_times, pred_times, f1_trains, f1_tests, gs_times)
```
Comment (parambharat) [inline app.py:20]: We need a common style guide at least for naming things.
Either `snake_case` or `camelCase` for function_names.
variables as `smallnocaps`
Classes as CapitalCase.

File: app.py [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
 import pandas as pd
-from sklearn import grid_search
-from sklearn.cross_validation import train_test_split
 
 from lib.data.wrangler import readData
 from lib.helpers import runTests, writeToCsv
 
-X_test, \
-    X_train, \
+X_train, \
```
Comment (joshuacook) [inline app.py:8]: I'm having to reverse these when I run it. 

Is there a way to specify by key and not by index?

Comment (dtimm) [reply app.py:8]: That's no good. I had to reverse on my end to get it to run... Is the pickling process deterministic?

Regardless, it should just check that the columns return in the correct order (probably the initial order).

Comment (joshuacook) [inline app.py:8]: @dtimm I know you didn't create this problem so probably not fair on your PR. Just noticed.

File: lib/model/__init__.py [hunk 3]
```diff
@@ -1,33 +1,72 @@
 import time
 from sklearn.metrics import f1_score
+from sklearn.grid_search import GridSearchCV
 
 class Model:
-    def __init__(self, classifier, parameters={}):
-        self.classifier = classifier(**parameters)
+    def __init__(self, classifier, parameters={}, gs_params={}):
+        self.classifier = classifier
+        self.model = classifier(**parameters)
```
Comment (joshuacook) [inline lib/model/__init__.py:8]: +1 Is true ... the classifier is the classifier and the model is the model. Nice!

File: lib/model/classifiers.py [hunk 4]
```diff
@@ -41,7 +41,7 @@
     # semi_supervised.LabelPropagation,
     # svm.LinearSVC,
     # svm.NuSVC,
-    (svm.SVC, {}),
-    (tree.DecisionTreeClassifier, {'random_state' : 42})
+    (svm.SVC, {}, {}),
+    (tree.DecisionTreeClassifier, {'random_state' : 42}, {})
```
Comment (joshuacook) [inline lib/model/classifiers.py:45]: the empty GS will need a random_state also

**Pull request merged**